### DISCUSSION TITLE
cache.updateQuery uses old cache in subscriptions

### DISCUSSION THREAD

**Discussion by gopeter** [urql-graphql/urql#2670] [Q&A] [answered]
# Update

Seems to be an error on my side, where a query ran in parallel. I'll update the discussion soon

# Original question

I'm trying to combine a query and a subscription to fetch a list of comments and add/update/delete comments on the fly with a subscription:

```tsx
const cacheExchange = createCacheExchange({
  updates: {
    Subscription: {
      ticketStream: (subscriptionResult, args, cache, info) => {
        cache.updateQuery({ query: GetTicketStreamDocument, variables: { id: (args as SubscribeTicketStreamSubscriptionVariables).ticketId } }, (data) => {
          data.ticket.stream.nodes.push(subscriptionResult)
          return data
        })
      }
    }
  }
})
```

This works once, like this:

- Comment A (from initial query)
- Comment B (from initial query)
- Comment C (from subscription)

But once I get more than one new record from the subscription, it always uses the old cache und just appends the newest one to the initial list:

- Comment A (from initial query)
- Comment B (from initial query)
- Comment D (from subscription)

instead of

- Comment A (from initial query)
- Comment B (from initial query)
- Comment C (from subscription)
- Comment D (from subscription)

What can I do to write to the latest cache of the GetTicketStreamQuery?

**Comment by JoviDeCroock** [comment, marked as answer]
@gopeter anything we can do here?

**Reply by gopeter** [reply]
Sorry, I should have already closed this one. It was an error on my side, because I refetched records while using a subscription which then lead to weird cache behaviors.
